### ISSUE TITLE
Unexpected function return value warning in a while loop

### ISSUE THREAD

**Issue by YorigamiShion** [Microsoft/pyright#45]
**Describe the bug**
When a function return in a while True loop, pyright will report 'Function with declared type of (declared return type) must return value'.

**To Reproduce**
After pyright checked the code, it report this problem.

**Expected behavior**
Do not report this.

**Screenshots or Code**
```python
def f(length: int) -> int:
    n = 0
    while True:
        if n >= length:
            return n
        n += random.randint(1, 5) 
```

**VS Code extension or command-line**
VS Code extension in 1.0.8

**Comment by erictraut** [comment]
Yeah, it's probably reasonable to expect the type analyzer to recognize this special case since `while True` is a commonly-used pattern. In the general case, the analyzer will need to assume that the code flow will eventually exit the `while` loop and return `None`.

If you want to work around this in the meantime, you can add a `return 0` statement below your while loop.

**Comment by erictraut** [comment]
I've added code to the analyzer to handle while loops with constant conditionals. This functionality will be in the next published version (1.0.9). Thanks for the bug report.

**Comment by erictraut** [comment]
This is fixed in 1.0.9, which I just published.
